$OpenBSD: patch-plugins_decoder_ffmpeg_k3bffmpegwrapper_cpp,v 1.5 2018/04/22 10:25:33 rsadowski Exp $

commit 1777236203f21eed7a9baade632472094c8081d3
Author: Pino Toscano <pino@kde.org>
Date:   Sat Feb 4 10:48:45 2017 +0100

    ffmpeg: fix/simplify metadata conversion to string

    Comparing a pointer with an integer value is (correctly) an error with
    GCC 7.

Index: plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.orig
+++ plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -240,12 +240,7 @@ QString K3bFFMpegFile::title() const
 {
     // FIXME: is this UTF8 or something??
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "TITLE", NULL, 0 );
-    if( ade == NULL )
-        return QString();
-    if( ade->value != '\0' )
-        return QString::fromLocal8Bit( ade->value );
-    else
-        return QString();
+    return ade && ade->value && ade->value[0] != '\0' ? QString::fromLocal8Bit( ade->value ) : QString();
 }
 
 
@@ -253,12 +248,7 @@ QString K3bFFMpegFile::author() const
 {
     // FIXME: is this UTF8 or something??
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "ARTIST", NULL, 0 );
-    if( ade == NULL )
-        return QString();
-    if( ade->value != '\0' )
-        return QString::fromLocal8Bit( ade->value );
-    else
-        return QString();
+    return ade && ade->value && ade->value[0] != '\0' ? QString::fromLocal8Bit( ade->value ) : QString();
 }
 
 
@@ -266,12 +256,7 @@ QString K3bFFMpegFile::comment() const
 {
     // FIXME: is this UTF8 or something??
     AVDictionaryEntry *ade = av_dict_get( d->formatContext->metadata, "COMMENT", NULL, 0 );
-    if( ade == NULL )
-        return QString();
-    if( ade->value != '\0' )
-        return QString::fromLocal8Bit( ade->value );
-    else
-        return QString();
+    return ade && ade->value && ade->value[0] != '\0' ? QString::fromLocal8Bit( ade->value ) : QString();
 }
 
 
